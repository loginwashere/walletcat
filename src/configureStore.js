import { createStore, applyMiddleware, compose } from 'redux'
import reducers from './reducers'
import middleware from './middleware'
import { addLocaleData } from 'react-intl'
import { DEFAULT_LOCALE } from './config'
import localeMessages from './translations'

import enLocaleData from 'react-intl/locale-data/en'
import ruLocaleData from 'react-intl/locale-data/ru'
import ukLocaleData from 'react-intl/locale-data/uk'

addLocaleData(enLocaleData)
addLocaleData(ruLocaleData)
addLocaleData(ukLocaleData)

const locale = DEFAULT_LOCALE
const messages = localeMessages[locale]

const initialState = {
  intl: {
    locale,
    messages,
  }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const configureStore = (preloadedState = {}) => createStore(
  reducers,
  {
    ...preloadedState,
    ...initialState
  },
  composeEnhancers(applyMiddleware(...middleware))
)

export default configureStore
