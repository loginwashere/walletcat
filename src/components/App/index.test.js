import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { App } from '.'
import configureStore from '../../configureStore'

it('renders without crashing', () => {
  const store = configureStore()
  const tree = renderer.create(
    <Provider store={store} >
      <App dispatch={store.dispatch}
           isAuthenticated={false}
           location={{ pathname: '/sign-in' }} />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders landing without crashing', () => {
  const store = configureStore()
  const div = document.createElement('div')
  document.body.appendChild(div)
  ReactDOM.render(
    <Provider store={store} >
      <App dispatch={store.dispatch}
           isAuthenticated={false}
           location={{ pathname: '/' }} />
    </Provider>,
    div
  )
})
