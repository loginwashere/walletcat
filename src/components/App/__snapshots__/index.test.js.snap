exports[`components:App: renders without crashing 1`] = `
<div>
  <nav
    className="navbar navbar-default navbar-fixed-top">
    <div
      className="container">
      <div
        className="navbar-header">
        <a
          className="header__brand__link navbar-brand"
          onClick={[Function]}
          style={Object {}}>
          <img
            className="header__brand__image"
            height={50}
            src="/safari-pinned-tab.svg"
            width={50} />
          <span
            className="header__brand__name">
            Wallet Cat
          </span>
        </a>
        <button
          className="navbar-toggle collapsed"
          onClick={[Function]}
          type="button">
          <span
            className="sr-only">
            Toggle navigation
          </span>
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
        </button>
      </div>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse">
        <ul
          className="nav navbar-nav"
          role={null} />
        <ul
          className="nav navbar-nav navbar-right"
          role={null}>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button">
              Sign in
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button">
              Create an account
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="appContent container">
    <div />
  </div>
  <nav
    className="navbar navbar-default navbar-fixed-bottom">
    <div
      className="container">
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse">
        <ul
          className="nav navbar-nav"
          role={null}>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button">
              About
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button">
              Contact Us
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button">
              Privacy Policy
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              action="push"
              href="#"
              onClick={[Function]}
              role="button">
              Terms and conditions
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
